packages/blade/src/components/Input/BaseInput/baseInputStyles.ts

```diff
@@ -46,60 +54,75 @@ export const getInputBackgroundAndBorderStyles = ({
   | 'isDropdownTrigger'
 >): CSSObject => {
   // normal state
-  let backgroundColor = theme.colors.interactive.background.gray.default;
-  let borderBottomColor = theme.colors.interactive.background.gray.highlighted;
+  let backgroundColor = getIn(theme.colors, baseInputBackgroundColor.default);
+  let borderColor = getIn(theme.colors, baseInputBorderColor.default);
+  let borderWidth = getIn(theme.border.width, baseInputBorderWidth.default);
 
-  // hoverState
-  if (isHovered) {
-    backgroundColor = theme.colors.interactive.background.gray.highlighted;
-  }
+  const baseInputState = isHovered
+    ? 'hovered'
+    : isFocused
+    ? 'focused'
+    : isDisabled
+    ? 'disabled'
+    : 'default';
```

Comment:
move it to `getInputState({isHovered, isFocussed, isDisabled})` function with if-else maybe?

Reply:
Done. Created a common function